sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogManager.scala

```diff
@@ -106,13 +106,14 @@ class CatalogManager(
   }
 
   def setCurrentNamespace(namespace: Array[String]): Unit = synchronized {
-    if (currentCatalog.name() == SESSION_CATALOG_NAME) {
-      if (namespace.length != 1) {
+    currentCatalog match {
+      case _ if currentCatalog.name() == SESSION_CATALOG_NAME && namespace.length == 1 =>
+        v1SessionCatalog.setCurrentDatabase(namespace.head)
+      case catalog: SupportsNamespaces if catalog.namespaceExists(namespace) =>
+        logInfo(s"set current namespace to ${namespace.mkString(".")}")
+        _currentNamespace = Some(namespace)
+      case _ =>
         throw new NoSuchNamespaceException(namespace)
```

Comment:
It's very likely that the underlying catalog can't report namespace existence directly, but can throw namespace not found error when calling APIs like `loadTable(ident)`

Reply:
And we should allow users to set current namespace when working with these catalogs.

Reply:
> It's very likely that the underlying catalog can't report namespace existence directly, but can throw namespace not found error when calling APIs like `loadTable(ident)`

First, the user needs to sense whether the namespace exists.
Secondly, Those who cannot obtain the namespace status can return true directly in namespaceExists.
Finally, for those don't support namespaces, users should be allowed to set the defaultNamespace, I will add this.